core/templates/pages/topic-editor-page/modal-templates/topic-editor-save-modal.template.html

```diff
@@ -5,14 +5,15 @@ <h3>
   </h3>
 </div>
 <div class="modal-body oppia-long-text">
-  <p>
+  <p ng-if="isTopicPublished">
     Please enter a brief description of what you have changed:
-    <textarea rows="3" cols="50" maxlength="<[MAX_COMMIT_MESSAGE_LENGTH]>" ng-model="commitMessage" focus-on="saveChangesModalOpened" class="protractor-test-commit-message-input" autofocus></textarea>
+    <textarea rows="3" cols="50" maxlength="<[MAX_COMMIT_MESSAGE_LENGTH]>" ng-model="commit.message" focus-on="saveChangesModalOpened" class="protractor-test-commit-message-input" autofocus></textarea>
   </p>
+  <p ng-if="!isTopicPublished">Do you want to save the changes?</p>
 </div>
 <div class="modal-footer">
   <button class="btn btn-secondary" ng-click="cancel()" type="button">Cancel</button>
-  <button class="btn btn-success protractor-test-close-save-modal-button" ng-disabled="!commitMessage" ng-click="confirm(commitMessage)">
+  <button class="btn btn-success protractor-test-close-save-modal-button" ng-disabled="!commit.message && isTopicPublished" ng-click="confirm(commit.message||'commit-before-publishing')">
```

Comment:
Why not just pass an empty message when the topic is not published?

Reply:
Hm actually thinking a bit more about this, I think it would be better to show the textarea all the time but only make it mandatory to fill in when the topic is about to be published.

Reply:
Done!